src/components/AreaChart.tsx

```diff
@@ -30,23 +31,29 @@ type IAreaChartNonBucketProps = Subtract<IAreaChartProps, IAreaChartBucketProps>
  * is a component with bucket props measures, viewBy, stacksBy, filters
  */
 export function AreaChart(props: IAreaChartProps): JSX.Element {
+    showWarningForAreaChart(props);
+    const areaChartProps = getPropsForArea(props);
```

Comment:
hmm this also is doing too much things :/ we are already in AreaChart component so calling using component name inside this component is redundant information for me...

creating such a this generic method is making tests complex or even difficult to test as its forcing you to mock a lot of data... so if there is a way to split the context to handle less conditions that its better :) 

my suggestion:
`getBucketsProps`, `getConfigProps` :)